File: python/007.py
"""
Project Euler - Problem Solution 007
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def get_nth_prime(nth):
  limit = nth
  primes, composites = set(), set()
  num_primes = 0
  while(num_primes < nth):
    limit *= 2 # increase the limit if there are less than nth primes
    for i in range(2, limit+1):
      if i not in composites:
        if i not in primes:
          primes.add(i)
          if(len(primes) == nth):
            return i
        for j in range(i*i, limit+1, i):
          composites.add(j)
    num_primes = len(primes)

if __name__ == "__main__":
  print(get_nth_prime(10001))

File: javascript/007.js
/**
 * Project Euler - Problem Solution 007
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

const getNthPrime = (nth) => {
  let limit = nth;
  const primes = new Set(), composites = new Set();
  let num_primes = 0;
  while(num_primes < nth) {
    limit *= 2;
    for(let i = 2; i < limit+1; i++) {
      if(!composites.has(i)) {
        if(!primes.has(i)) {
          primes.add(i);
          if(primes.size === nth) {
            return i;
          }
        }
        for(let j = i*i; j < limit+1; j += i) {
          composites.add(j);
        }
      }
    }
    num_primes = primes.size;
  }
};

console.log(getNthPrime(10001));